### PR TITLE
ci: Update labeler.yml with additional files/labels

### PR DESCRIPTION
**Pull request by nathan-weinberg**
Auto-labeler hasn't been updated in a bit, making some additions here with existing files/labels for maintainer ease

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -1,16 +1,40 @@
...
           - CODE_OF_CONDUCT.md
+          - CONTRIBUTOR_RULES.md
+          - governance.md
+          - MAINTAINERS.md
           - README.md
+          - SECURITY.md
           - TROUBLESHOOTING.md
           - CONTRIBUTING/**/*
           - docs/**/*
+          - MAINTAINERS/**/*
```
Comment (russellb) [inline .github/labeler.yml:33]: It looks like this supports globbing, so `*.md` may be better for the top level?

File: .github/labeler.yml [hunk 2]
```diff
@@ -1,16 +1,40 @@
 # SPDX-License-Identifier: Apache-2.0
 
+CI/CD:
+  - changed-file:
+      - any-glob-to-any-file:
+          - .github/labeler.yml
```
Comment (russellb) [inline .github/labeler.yml:6]: How about `.github/*`

Comment (nathan-weinberg) [reply .github/labeler.yml:6]: `PULL_REQUEST_TEMPLATE.md` shouldn't be included IMO

File: .github/labeler.yml (hunk 1 above)
Comment (nathan-weinberg) [reply .github/labeler.yml:33]: Thought about that, but some Markdown files aren't docs necessarily, e.g. `PULL_REQUEST_TEMPLATE.md`

**Review by russellb — APPROVED** [review]

**Pull request merged**